jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -802,7 +802,20 @@ ecma_op_function_get_function_realm (ecma_object_t *func_obj_p) /**< function ob
     const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
     return ecma_op_function_get_realm (bytecode_data_p);
   }
-
+#if ENABLED (JERRY_BUILTIN_PROXY)
+  if (ECMA_OBJECT_IS_PROXY (func_obj_p))
+  {
+    ecma_proxy_object_t *proxy_obj_p = (ecma_proxy_object_t *) func_obj_p;
+    if (ecma_is_value_null (proxy_obj_p->handler))
+    {
+      ecma_raise_type_error (ECMA_ERR_MSG ("Prototype from revoked Proxy is invalid."));
+      return NULL;
+    }
+    // ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) ecma_get_object_from_value (proxy_obj_p->target);
+    // const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_obj_p);
+    return ecma_op_function_get_function_realm (ecma_get_object_from_value (proxy_obj_p->target));
+  }
+#endif /* ENABLED (JERRY_BUILTIN_PROXY) */
   JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION);
```

Comment:
I would move this code below inside an `if`. Furthermore I would convert the whole thing into a while loop, maybe it could work in the bound function case. Suggested structure of the code:

```
while (true) {
  if (func is byte_code) return ((byte_code)func)->realm;
  if (func is native_function) return ((native_func)func)->realm
  if (func is proxy) {
    func = proxy->target
    continue
  }
  assert(func is bound_function)
  func = bound_target
}
```